sdk/core/core-rest-pipeline/test/defaultRetryPolicy.spec.ts

```diff
@@ -57,4 +57,27 @@ describe("defaultRetryPolicy", function () {
     assert.strictEqual(next.callCount, DEFAULT_RETRY_POLICY_COUNT + 1);
     assert.isTrue(catchCalled);
   });
+
+  it("It should not retry on RestError with status 416", async () => {
+    const request = createPipelineRequest({
+      url: "https://bing.com",
+    });
+    const testError = new RestError("Test Error!", { statusCode: 416 });
+
+    const policy = defaultRetryPolicy();
+    const next = sinon.stub<Parameters<SendRequest>, ReturnType<SendRequest>>();
+    next.rejects(testError);
+
+    const clock = sinon.useFakeTimers();
+
+    let catchCalled = false;
+    const promise = policy.sendRequest(request, next);
+    promise.catch((e) => {
+      catchCalled = true;
+      assert.strictEqual(e, testError);
+    });
+    await clock.runAllAsync();
+    assert.strictEqual(next.callCount, 1);
+    assert.isTrue(catchCalled);
```

Comment:
nit: I think this could be rewritten more concisely along the lines of:
```suggestion
    await assert.isRejected(policy.sendRequest(request, next).then(() => clock.runAllAsync()), /Test Error/);
    assert.strictEqual(next.callCount, 1);
```

Reply:
also could the test be simplified by dropping using fake timers?

Reply:
Thank you, Deya! I have simplified the tests. I removed the fake timers on the requests that were expected to throw the first time, but I kept the fake timers on the test that expected retries to happen.